ipa-core/src/protocol/basics/reveal.rs

```diff
@@ -190,7 +198,7 @@ impl<'a, F: ExtendableField> Reveal<UpgradedMaliciousContext<'a, F>, RecordId>
             .await?;
 
             if share_from_left == share_from_right {
-                Ok(Some(left + right + share_from_left))
+                Ok(Some((left + right + share_from_left).into_array()))
```

Comment:
we probably want to use this version of malicious reveal, so it would be worth covering this line with tests